db/view/view_update_generator.cc

```diff
@@ -33,28 +34,46 @@ future<> view_update_generator::start() {
             if (_sstables_with_tables.empty()) {
                 _pending_sstables.wait().get();
             }
-            while (!_sstables_with_tables.empty()) {
-                auto& [sst, t] = _sstables_with_tables.front();
+            for (auto& [t, sstables] : _sstables_with_tables) {
                 try {
                     schema_ptr s = t->schema();
-                    flat_mutation_reader staging_sstable_reader = sst->read_rows_flat(s, no_reader_permit());
-                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, _db, sst, _as), db::no_timeout);
+                    auto ssts = make_lw_shared(t->get_compaction_strategy().make_sstable_set(s));
+                    for (auto& sst : sstables) {
+                        ssts->insert(sst);
+                    }
+
+                    flat_mutation_reader staging_sstable_reader =
+                        ::make_range_sstable_reader(s,
+                                                    no_reader_permit(),
+                                                    std::move(ssts),
+                                                    query::full_partition_range,
+                                                    s->full_slice(),
+                                                    service::get_local_streaming_read_priority(),
+                                                    nullptr,
+                                                    ::streamed_mutation::forwarding::no,
+                                                    ::mutation_reader::forwarding::no);
+
+                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, *t, sstables, _as), db::no_timeout);
                     if (result == stop_iteration::yes) {
                         break;
                     }
                 } catch (...) {
+                    for (auto& sst : sstables) {
                     vug_logger.warn("Processing {} failed: {}. Will retry...", sst->get_filename(), std::current_exception());
```

Comment:
I'm fine with printing all the sstables names that we didn't process, but seems misleading and redundant to print `current_exception` for each one. It will bloat the log and bring false information, since technically the error occurred only when processing one of the sstables, not for every single one.

Reply:
Note that this is temporary. When we move to a compaction, the whole compaction will fail and we'll have the messages coming from compactions.